### PR TITLE
Prefix type docs with the object kind (archetype, component, datatype)

### PR DESCRIPTION
**Pull request by emilk**
### What
* Closes https://github.com/rerun-io/rerun/issues/3540

image

image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3616) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3616)
- [Docs preview](https://rerun.io/preview/6fe84cecc0d594fd5cf1583bc4923ced7f08e9ad/docs) 
- [Examples preview](https://rerun.io/preview/6fe84cecc0d594fd5cf1583bc4923ced7f08e9ad/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: crates/re_types/definitions/rerun/components/keypoint_id.fbs [hunk 1]
```diff
@@ -10,14 +10,14 @@ namespace rerun.components;
 
 /// A 16-bit ID representing a type of semantic keypoint within a class.
 ///
-/// \py `KeypointId`s are only meaningful within the context of a [`rerun.components.ClassDescription`][].
+/// \py `KeypointId`s are only meaningful within the context of a [`rerun.datatypes.ClassDescription`].
```
Comment (Wumpf) [inline crates/re_types/definitions/rerun/components/keypoint_id.fbs:13]: aren't those now under the `rerun` namespace directly?

File: crates/re_types_builder/src/codegen/python.rs [hunk 2]
```diff
@@ -692,7 +685,17 @@ fn code_for_struct(
 
             code.push_text(format!("{name}: {typ}"), 1, 4);
 
-            code.push_text(quote_doc_from_docs(docs), 0, 4);
+            // Generating docs for all the fields creates A LOT of visual noise in the API docs.
+            let show_fields_in_docs = false;
+            if show_fields_in_docs {
+                code.push_text(quote_field_docs(field), 0, 4);
+            } else {
+                code.push_text(
```
Comment (Wumpf) [inline crates/re_types_builder/src/codegen/python.rs:693]: if we make the effort to leave a comment that the doc string isn't there, we might as well put the commented out docstring in here 🤔 - after all someone just went to look at the code now and wants to know more about the field.

**Review by Wumpf — APPROVED** [review]
lgtm - nits

File: crates/re_types/definitions/rerun/components/keypoint_id.fbs (hunk 1 above)
Comment (emilk) [reply crates/re_types/definitions/rerun/components/keypoint_id.fbs:13]: We do import some things to the main `rerun` namespace, but I think for doclinks it is better to be explicit like htis

**Pull request merged**